src/audio/drc/drc.c

```diff
@@ -156,19 +156,17 @@ static struct comp_dev *drc_new(const struct comp_driver *drv,
 	if (bs > SOF_DRC_MAX_SIZE) {
 		comp_cl_err(&comp_drc, "drc_new(), error: configuration blob size = %u > %d",
 			    bs, SOF_DRC_MAX_SIZE);
-		return NULL;
+		goto fail;
 	}
 
 	dev = comp_alloc(drv, sizeof(*dev));
 	if (!dev)
-		return NULL;
+		goto fail;
```

Comment:
same here since it is only the first alloc

Reply:
done, exit paths:
* `return NULL` = dev not allocated
* `goto fail` = dev allocated
* `goto cd_fail` = cd allocated